test/integration/ynEIGEN/TokenStakingNode.t.sol

```diff
@@ -3,37 +3,43 @@ pragma solidity 0.8.24;
 
 import {ynEigenIntegrationBaseTest} from "test/integration/ynEIGEN/ynEigenIntegrationBaseTest.sol";
 import {IERC20} from "lib/openzeppelin-contracts/contracts/interfaces/IERC20.sol";
```

Comment:
Can we add tests that are equivalent to this:

[testDelegateUndelegateAndDelegateAgainWithExistingStake](https://github.com/yieldnest/yieldnest-protocol/blob/f16c9cab5c5858106b3b12d4d8f05d135ef7954c/test/integration/StakingNode.t.sol#L665)

https://github.com/yieldnest/yieldnest-protocol/blob/f16c9cab5c5858106b3b12d4d8f05d135ef7954c/test/integration/StakingNode.t.sol#L704C14-L704C64